Unbox.swift

```diff
@@ -674,34 +666,78 @@ private extension Unboxer {
+        }
+    }
+
+    static func performUnboxingWithData<T: Unboxable>(data: NSData, context: Any? = nil, allowInvalidElements: Bool = false) throws -> T {
+        return try Unboxer.unboxerFromData(data, context: context).performUnboxing(allowInvalidElements)
+    }
+
+    static func performUnboxingWithData<T: Unboxable>(data: NSData, context: Any? = nil, allowInvalidElements: Bool = false) throws -> [T] {
+        return try Unboxer.unboxersFromData(data, context: context).map {
+            return try $0.performUnboxing(allowInvalidElements)
+        }
+    }
+
+    static func performUnboxingWithDictionaryAndContext<T: UnboxableWithContext>(dictionary: UnboxableDictionary, context: T.ContextType, allowInvalidElements: Bool = false) throws -> T {
+        return try Unboxer(dictionary: dictionary, context: context).performUnboxingWithContext(context, allowInvalidElements: allowInvalidElements)
+    }
+
+    static func performUnboxingWithDictionariesAndContext<T: UnboxableWithContext>(dictionaries: [UnboxableDictionary], context: T.ContextType, allowInvalidElements: Bool = false) throws -> [T] {
+        return try dictionaries.map {
+            try performUnboxingWithDictionaryAndContext($0, context: context, allowInvalidElements: allowInvalidElements)
+        }
+    }
+
+    static func performUnboxingWithDataAndContext<T: UnboxableWithContext>(data: NSData, context: T.ContextType, allowInvalidElements: Bool = false) throws -> T {
+        return try Unboxer.unboxerFromData(data, context: context).performUnboxingWithContext(context, allowInvalidElements: allowInvalidElements)
+    }
+
+    static func performUnboxingWithDataAndContext<T: UnboxableWithContext>(data: NSData, context: T.ContextType, allowInvalidElements: Bool = false) throws -> [T] {
+        return try Unboxer.unboxersFromData(data, context: context).map {
+            return try $0.performUnboxingWithContext(context, allowInvalidElements: allowInvalidElements)
+        }
+    }
     
-    func performUnboxing<T: Unboxable>() throws -> T {
+    func performUnboxing<T: Unboxable>(allowInvalidElements: Bool) throws -> T {
         let unboxed = T(unboxer: self)
-        try self.throwIfFailed()
+        try self.throwIfFailed(allowInvalidElements)
         return unboxed
     }
     
-    func performUnboxingWithContext<T: UnboxableWithContext>(context: T.ContextType) throws -> T {
+    func performUnboxingWithContext<T: UnboxableWithContext>(context: T.ContextType, allowInvalidElements: Bool) throws -> T {
         let unboxed = T(unboxer: self, context: context)
-        try self.throwIfFailed()
+        try self.throwIfFailed(allowInvalidElements)
         return unboxed
     }
     
-    func performCustomUnboxingWithClosure<T>(closure: Unboxer -> T?) throws -> T {
+    func performCustomUnboxingWithClosure<T>(allowInvalidElements: Bool, closure: Unboxer -> T?) throws -> T {
```

Comment:
Right, it's just syntax sugar for internals, which we even aren't using since we're passing variables as parameters anyway. I was only thinking in case someone wanted to extend the library and has that available, but didn't feel strongly about it. Method names matching the 1st parameter is a good convention I wasn't thinking about :+1: